pkg/gofr/serrors/error.go

```diff
@@ -0,0 +1,129 @@
+package serrors
+
+import (
+	"fmt"
+	"time"
+)
+
+const UNSET = "NA"
+
+// New creates a new Error instance by wrapping an existing error with additional context.
+// It captures the current timestamp in nanoseconds for precise error tracing and sets default values
+// for error metadata. The resulting error maintains the original error chain while providing a new
+// contextual message.
+//
+// Parameters:
+//
+//	err     : The original error being wrapped (can be nil). Preserved for error chain inspection.
+//	message : Human-readable context message describing the error situation. Should provide
+//	          higher-level context about where/why the error occurred.
+//
+// Returns:
+//
+//	*Error : Structured error instance containing:
+//	         - Wrapped cause (original error)
+//	         - Provided contextual message
+//	         - Timestamp in UTC nanoseconds (meta.timestamp)
+//	         - Default metadata:
+//	             statusCode: "NA" (unclassified status)
+//	             subStatusCode: "NA" (unclassified sub-status)
+//	             level: ERROR (default severity)
+//	             retryable: false (default non-retryable)
+//	             externalStatusCode: 0 (unset external status)
+//	             externalMessage: "NA" (unset client message)
+//
+// Usage:
+//
+//	if err := io.Read(file); err != nil {
+//	    return errors.Wrap(err, "config load failed")
+//	}
+func New(err error, message string) *Error {
+	meta := make(map[string]any)
+
+	// Capture precise timestamp for error tracing
+	meta["timestamp"] = fmt.Sprintf("%d", (time.Now().UTC()).UnixNano())
```

Comment:
```suggestion
	meta["timestamp"] = fmt.Sprint(time.Now().UnixNano())
```

Reply:
I initially made this more explicit to help catch any underlying changes. On reflection, it might be more than required, so I'm considering simplifying it